include/cotila/matrix/math.h

```diff
@@ -64,45 +64,62 @@ constexpr matrix<T, M, P> matmul(const matrix<T, M, N> &a,
   return generate<M, P>([&a, &b](auto i, auto j){return cotila::sum(a.row(i)*b.column(j));});
 }
 
-/** @brief computes the matrix inverse 
- *  @param m an \f$ M \times M \f$ matrix of type T
- *  @return The inverse of \f$ \textbf{M} \f$, \f$ \textbf{M}^{-1}\f$ such that 
- *  \f$ \textbf{M}\textbf{M}^{-1} == \textbf{M}^{-1}\textbf{M} == \textbf{I}_{M} \f$
+
+/** @brief Perform Gaussian Elimination
+ *  @param _a an \f$ M \times M \f$ matrix of type T
+ *  @param _adj A \f$ M \times N \f$ matrix 
+ *  @return b, A \f$ M \times N \f$ matrix x that solves the equation
+ *  \f$ \textbf{A}\textbf{x} = \textbf{b} \f$.
  *
- *  Computes the inverse of a matrix..
+ *  Perform Gaussian elimination.
  */
-template <typename T, std::size_t M>
-constexpr matrix<T, M, M> inverse(const matrix<T, M, M> &m) {
-  matrix<T, M, M> inv = identity<T, M>;
-  auto A = m;
+template <typename T, std::size_t M, std::size_t N>
+constexpr matrix<T, M, N> gauss_elim(const matrix<T, M, M> &_a,
+                                     const matrix<T, M, N> &_adj) {
+  auto a = _a;
```

Comment:
what's going on here?

Reply:
_a and _adj are const so I copied them. Guess I coulda justed passed them in as a copy. Shrug. Thoughts?